projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+type RequestError struct {
+	SleepTime int
+	Retry     bool
+	ErrMsg       error
+}
+
+
+
+func parseRequest(resp http.Response) (reqBody string, reqError RequestError) {
+
+	switch resp.StatusCode {
+	case 429:
+		retryTime := resp.Header.Get("Retry-After")
+		formattedRetryTime, err := ParseRetryValue(retryTime)
+
+		if err != nil {
+			// cannot determine how long to sleep for, give up
+			reqError.ErrMsg = errors.New("cannot get weather at the moment try again later")
+			return
+		}
+		if formattedRetryTime > 5 {
+			// sleep time is long give up
+			reqError.ErrMsg = errors.New("server is busy at the moment try again later")
+
+		} else {
+			reqError.ErrMsg = errors.New("server is busy, retrying in few seconds")
+			reqError.SleepTime = formattedRetryTime
+			reqError.Retry = true
+		}
+	default:
+		body, _ := io.ReadAll(resp.Body)
```

Comment:
You shouldn't ignore this error - if you couldn't read the body, you should probably report that something went wrong.